### PR TITLE
Pull request info

### PR DESCRIPTION
**Pull request by Katsiez**
Link to my deployed project:
Coming up shortly in the README file. https://sharp-curran-cf688a.netlify.app/ 

My reflections on how this week's project turned out:
Difficult, I struggled a lot, cried, and drank a lot of wine. But it's okay, I still somehow managed to create a somewhat decent app.  

Things I'd like to have clarified or explained in more detail: 

For this project, I attempted to change my forecast and current weather URLs by replacing the city name to a variable called 'city' and the API key digits to 'apiKey', both previously defined. Each time I tried to do this, either forecast or current weather would rebel and not display the data.
I am therefore not expecting this app to be properly displayed on Netlify and this project will instead have to be run through running the index.html file.

-This is what I would like to clarify.

### CODE REVIEW

File: code/weather.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+//Fetch with JSON - Current Weather
+const weatherToday = () => {
+    const currentWeatherUrl = "https://api.openweathermap.org/data/2.5/weather?q=Toronto,Canada&units=metric&APPID=c2889b12ee617ea787319a19a98a5906"
+
+    fetch(currentWeatherUrl)
+        .then((response) => {
+            return response.json()
+        })
+        .then((json) => {
+            currentWeatherToday(json)
```
Comment (ankimelin) [inline code/weather.js:51]: I like that you call a function here!

File: code/weather.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            minMaxTemps[currentDate] = {
+                minTemp: item.main.temp_min,
+                maxTemp: item.main.temp_max,
+                dayOfWeek: weekDays[trueDate].toUpperCase(),
+                icon: item.weather[0].icon
+            }
+        }
+    })
+    const forecast = document.getElementById('weatherForecast')
+    const forecastIcon = document.getElementById('forecastIcon')
```
Comment (ankimelin) [inline code/weather.js:117]: Should these also have been declared at the top?

File: code/weather.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        forecast.innerHTML += `<div class="column">${minMaxTemps[date].dayOfWeek}</div>`
+        forecast.innerHTML += `<div class="column">${minMaxTemps[date].minTemp.toFixed(0.5)} °C|${minMaxTemps[date].maxTemp.toFixed(0.5)} °C </div>`
+        forecast.innerHTML += `<img src="http://openweathermap.org/img/wn/${minMaxTemps[date].icon}@2x.png"></img>`
+    }
+}
+// Fetch 5 day Forecast Data
+const weatherForecastUrl = "https://api.openweathermap.org/data/2.5/forecast?q=Toronto,Canada&units=metric&APPID=c2889b12ee617ea787319a19a98a5906"
+fetch(weatherForecastUrl)
+    .then((response) => { return response.json() })
+    .then((data) => { updateMinMaxTemps(data) })
```
Comment (ankimelin) [inline code/weather.js:130]: Should this be moved further up?

**Review by ankimelin — APPROVED** [review]
Would have loved to see the live weather app, but no link included yet - reviewed only the code and not the req. 

So cool you managed the changed background and adding a city! Love the small comments which makes it easy to understand the purpose of the code.

As always - great work!!

File: code/index.html [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+  <body>
+    <!-- MAIN SECTION, CURRENT WEATHER -->
+    <section class id="weather-wrap">
+      <h1>Toronto Forecast</h1>
+    </section>
+
+    <section class="main">
+      <img class="cn-img" src="./assets/cntower.png" height="100px" />
+      <div class id="city">
```
Comment (EmOlsz) [inline code/index.html:18]: No need for empty `class` attribute

File: code/index.html [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        <p>Toronto, ON</p>
+      </div>
+      <div class id="date"></div>
+      <p>Current temperature:</p>
+      <div class id="temperature"></div>
+      <p>Visibility:</p>
+      <div class id="description"></div>
+      <img id="visibilityIcon" style="width: 40px" src="" />
+      <div class="highLow">
```
Comment (EmOlsz) [inline code/index.html:27]: classes should be written in `kebab-case`, not `camelCase`

File: code/index.html [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        <p>H:</p>
+        <div class id="H"></div>
+        <p>L:</p>
+        <div id="L"></div>
+      </div>
+
+      <p>Feels Like:</p>
+      <div class id="feelsLike"></div>
+      <section class id="sunUpSunDown">
+        <p>Sunrise: <img src="./assets/sunrise.png" style="width: 20px" /></p>
```
Comment (EmOlsz) [inline code/index.html:37]: Width of an image should be defined in CSS, not in HTML

File: code/style.css [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+body {
+  font-family: "Roboto Mono", monospace;
+}
+
+h1 {
+  text-align: center;
+  font-size: 20px;
+  margin: 50px 50px;
```
Comment (EmOlsz) [inline code/style.css:16]: `margin: 50px` is shorthand

File: code/style.css [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+  border-radius: 20px;
+  align-items: center;
+  text-align: center;
+  width: 100%;
+  margin: auto;
+  padding: 20px;
+  font-size: 14px;
+}
+
+#temperature {
```
Comment (EmOlsz) [inline code/style.css:41]: We shouldn't style by id's at all, always use classes

File: code/style.css [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+.column {
+  display: flex;
+  flex-basis: 40%;
+  margin: auto;
+}
+
+/* CHANGING BACKGROUND BASED ON TEMP*/
+body.cold {
+  background-image: url(./assets/cold.jpg);
```
Comment (EmOlsz) [inline code/style.css:70]: It's more safe to wrap url around quotes - some browsers might have problem with reading path without them 

```
background-image: url("./assets/cold.jpg");
```

File: code/weather.js [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+    } if (weather == 'Haze') {
+        document.getElementById('visibilityIcon').src = 'assets/thunder.png'
+    }
+}
+//CHANGING BACKGROUND DEPENDING ON TEMP
+const changeBackground = (json) => {
+    if (json.main.temp.toFixed(0.5) <= 10 || json.main.feels_like.toFixed(0.5) <= 10) {
+        body.classList.add('cold')
+    } else if (json.main.temp.toFixed(0.5) >= 22 || json.main.feels_like.toFixed(0.5) >= 2) {
+
```
Comment (EmOlsz) [inline code/weather.js:86]: empty line

File: code/weather.js [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+    for (const date in minMaxTemps) {
+        const forecast = document.getElementById('weatherForecast')
+        forecast.innerHTML += `<div class="column">${minMaxTemps[date].dayOfWeek}</div>`
+        forecast.innerHTML += `<div class="column">${minMaxTemps[date].minTemp.toFixed(0.5)} °C|${minMaxTemps[date].maxTemp.toFixed(0.5)} °C </div>`
+        forecast.innerHTML += `<img src="http://openweathermap.org/img/wn/${minMaxTemps[date].icon}@2x.png"></img>`
+    }
+}
+// Fetch 5 day Forecast Data
+fetch(weatherForecastUrl)
+    .then((response) => { return response.json() })
```
Comment (EmOlsz) [inline code/weather.js:128]: You don't need to write `return` keyword and brackets if it's same line :

```
.then(response => response.json())
```

**Review by EmOlsz — CHANGES_REQUESTED** [review]
In general it's well done project, congrats Katarina! Still, there are some things you need to change. They are mostly bad habits you probably didn't get rid of, so consider them (from now on) as a super important thing to pay attention for.
Provide changes and I will approve your project!

**Comment by Katsiez** [comment]
And here is the other one!
https://github.com/Technigo/project-weather-app/compare/master...Katsiez:master
:)

**Review by EmOlsz — APPROVED** [review]

**Pull request closed without merging**